src/test/compile-fail/issue-21546.rs

```diff
@@ -8,8 +8,14 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
+// Also works as a test for #14564
+
 mod Foo { }
+//~^ NOTE first definition of type or module
+
 struct Foo;
+//~^ ERROR duplicate definition of type or module `Foo`
```

Comment:
given that [RFC 218] ("empty struct with braces") was accepted, it would probably be a good idea to have both this test and a separate one of non-empty tuple structs, since I expect the code paths for how the two are handled to diverge somewhat in the future.

[RFC 218]: https://github.com/rust-lang/rfcs/blob/master/text/0218-empty-struct-with-braces.md